commands/peers/peer-rpc-svc.go

```diff
@@ -78,173 +74,78 @@ func (p *PeerService) ValidateDelete(nc netctx.Context, args *PeerDeleteReq) (*P
-		log.WithFields(log.Fields{
-			"error": err,
-			"path":  etcdmgmt.ETCDEnvFile,
-			"key":   "ETCD_INITIAL_CLUSTER",
-			"val":   env.InitialCluster,
-		}).Error("Failed to write Environment variable in to etcd conf file")
-		return err
-	}
-
-	if _, err = fp.WriteString("ETCD_INITIAL_CLUSTER_STATE=" + env.ClusterState + "\n"); err != nil {
-		log.WithFields(log.Fields{
-			"error": err,
-			"path":  etcdmgmt.ETCDEnvFile,
-			"key":   "ETCD_INITIAL_CLUSTER_STATE",
-			"val":   env.ClusterState,
-		}).Error("Failed to write Environment variable in to etcd conf file")
-		return err
-	}
-	return nil
-}
-
-// storeETCDProxyConf will store etcd configuration for proxy etcd
-func storeETCDProxyConf(env *EtcdConfigReq) error {
-	utils.InitDir(etcdmgmt.ETCDConfDir)
-	fp, err := os.OpenFile(etcdmgmt.ETCDProxyFile, os.O_CREATE|os.O_WRONLY, 0666)
-	if err != nil {
-		log.WithFields(log.Fields{
-			"error": err,
-			"path":  etcdmgmt.ETCDProxyFile,
-		}).Error("Failed to open etcd proxy file")
-		return err
-	}
-	defer fp.Close()
-
-	if _, err = fp.WriteString("ETCD_INITIAL_CLUSTER=" + env.InitialCluster + "\n"); err != nil {
-		log.WithFields(log.Fields{
-			"error": err,
-			"path":  etcdmgmt.ETCDProxyFile,
-			"key":   "ETCD_INITIAL_CLUSTER",
-			"val":   env.InitialCluster,
-		}).Error("Failed to write configuration in to etcd proxy file")
-		return err
-	}
-	return nil
-}
-
 // ExportAndStoreETCDConfig will store & export etcd environment variable along
 // with storing etcd configuration
 func (p *PeerService) ExportAndStoreETCDConfig(nc netctx.Context, c *EtcdConfigReq) (*PeerGenericResp, error) {
 	var opRet int32
 	var opError string
 
-	// TODO: Fix error handling here. In all error cases, we set opRet and
-	// opError. This isn't propagated back to client as peer response.
+	newEtcdConfig, err := etcdmgmt.GetNewEtcdConfig()
+	if err != nil {
+		opRet = -1
+		opError = fmt.Sprintf("Could not fetch etcd configuration.")
+		goto Out
+	}
+
 	if !c.DeletePeer {
+		// This is an add peer request containing information about
+		// which cluster to join.
 		if c.Client == false {
-			// Exporting etcd environment variable
-			os.Setenv("ETCD_NAME", c.Name)
-			os.Setenv("ETCD_INITIAL_CLUSTER", c.InitialCluster)
-			os.Setenv("ETCD_INITIAL_CLUSTER_STATE", c.ClusterState)
-
-			// Storing etcd envioronment variable in
-			// etcdEnvFile (/var/lib/glusterd/etcdenv.conf) locally. So that upon
-			// glusterd restart we can restore these environment variable again
-			err := storeETCDEnv(c)
-			if err != nil {
-				opRet = -1
-				opError = fmt.Sprintf("Could not able to write etcd configuration")
-				log.WithField("error", err.Error()).Error("Could not able to write etcd configuration")
-				return nil, err
-			}
+			newEtcdConfig.InitialCluster = c.InitialCluster
+			newEtcdConfig.ClusterState = c.ClusterState
+			newEtcdConfig.Name = c.Name
+			newEtcdConfig.Dir = c.Name + ".dir"
+			// TODO: Store config locally for glusterd2 and etcd restarts
```

Comment:
I wanted to a release with the embedded etcd, but not having this is a blocker. Shouldn't be too hard do to.

I'll merge this PR, but I'll wait till store/restore of etcd config is done to make the release.